julearn/model_selection/available_searchers.py

```diff
@@ -5,14 +5,52 @@
 # License: AGPL
 
 from copy import deepcopy
-from typing import List, Optional
+from typing import List, Optional, Type, Union
 
 from sklearn.model_selection import GridSearchCV, RandomizedSearchCV
 
 from julearn.utils.logging import logger, raise_error, warn_with_log
 
 
-_available_searchers = {"grid": GridSearchCV, "random": RandomizedSearchCV}
+_available_searchers = {
+    "grid": {
+        "class": GridSearchCV,
+        "params_attr": "param_grid",
+    },
+    "random": {
+        "class": RandomizedSearchCV,
+        "params_attr": "param_distributions",
+    },
+}
+
+try:
+    from skopt import BayesSearchCV
+
+    _available_searchers["bayes"] = {
+        "class": BayesSearchCV,
+        "params_attr": "search_spaces",
+    }
+except ImportError:
+    from sklearn.model_selection._search import BaseSearchCV
+
+    class BayesSearchCV(BaseSearchCV):
+        """Dummy class for BayesSearchCV that raises ImportError.
+
+        This class is used to raise an ImportError when BayesSearchCV is
+        requested but scikit-optimize is not installed.
+
+        """
+
+        def __init__(*args, **kwargs):
+            raise ImportError(
+                "BayesSearchCV requires scikit-optimize to be installed."
+            )
+
+    _available_searchers["bayes"] = {
```

Comment:
Can this repeated assignment be done in a `finally` block? I'm unsure as the import in one and definition in other branch might not play well, but would be nice if it works.